config/internal/configsource/manager.go

```diff
@@ -480,10 +547,10 @@ func newErrUnknownConfigSource(cfgSrcName string) error {
 	}
 }
 
-// parseCfgSrc extracts the reference to a config source from a string value.
+// parseCfgSrcInvocation extracts the reference to a config source from a string value.
 // The caller should check for error explicitly since it is possible for the
 // other values to have been partially set.
```

Comment:
Is it possible to add an example showing what this function does? Also why is it returning params Parser and not parsed params?